### ISSUE TITLE
How can I use getZoom() when onZoomChanged

### ISSUE THREAD

**Issue by T-Flynn** [googlemap-react/googlemap-react#31]
This is child，state is undefined
```
const GoogleContainer = (props) => {
  const { state } = useContext(GoogleMapContext);
  return (
    <ReactGoogleMap
      zoom={4}
      center={{ lat: 37.09024, lng: -95.712891 }}
      opt={{
        scaleControl: false,
      }}
      onZoomChanged={() => { console.log(state); }}
    >
      {props.markersList && props.markersList.map(item => (
        item && item.location
        && (
          <OverlayView
            position={{ lng: item.location.x, lat: item.location.y }}
            key={item.plateNo}
            id={item.plateNo}
          >
            <MarkerContent
              name={item.plateNo}
              position={{ lng: item.location.x, lat: item.location.y }}
              id={item.plateNo}
              angle={item.detail ? (item.detail.heading ? item.detail.heading : 0) : 0}
              data={item}
              showcontent={props.showContent}
              speed={null}
            />
          </OverlayView>
        )
      ))}
    </ReactGoogleMap>
  );
};
```
```
function ReactGoogleMap(props) {
  const { children, zoom, center, opt } = props;
  return (
    <GoogleMapProvider>
      <MapBox
        apiKey=""
        opts={{
          center,
          zoom: zoom || 4,
          clickableIcons: false,
          mapTypeControl: false,
          fullscreenControl: false,
          streetViewControl: false,
          ...opt,
        }}
        style={{ width: '100%', height: '100%' }}
        clickableIcons={false}
        language="en"
        onZoomChanged={props.onZoomChanged}
      />
      {children}
    </GoogleMapProvider>
  );
}

export default ReactGoogleMap;
```

**Comment by lucifer1004** [comment]
**TL;DR** You can combine `useState` and `useEffect` to achieve this. Put the call to `state` in the body of `useEffect`.

I am working on a refactor of this package.

For a temporary workaround, you can do something like this:

```typescript

const GoogleContainer = (props) => {
  const { state } = useContext(GoogleMapContext);
  const { trigger, setTrigger } = useState(0);
  useEffect(() => {
    if (state.map) console.log(state)
  }, [trigger])
  return (
    <ReactGoogleMap
      zoom={4}
      center={{ lat: 37.09024, lng: -95.712891 }}
      opt={{
        scaleControl: false,
      }}
      onZoomChanged={() => { setTrigger(trigger => trigger + 1) }}
    >
      {props.markersList && props.markersList.map(item => (
        item && item.location
        && (
          <OverlayView
            position={{ lng: item.location.x, lat: item.location.y }}
            key={item.plateNo}
            id={item.plateNo}
          >
            <MarkerContent
              name={item.plateNo}
              position={{ lng: item.location.x, lat: item.location.y }}
              id={item.plateNo}
              angle={item.detail ? (item.detail.heading ? item.detail.heading : 0) : 0}
              data={item}
              showcontent={props.showContent}
              speed={null}
            />
          </OverlayView>
        )
      ))}
    </ReactGoogleMap>
  );
};

```

**Comment by T-Flynn** [comment]
OK,I'll try this. Thanks

**Comment by T-Flynn** [comment]
Hi, I have tried this but it still not work, the state is always undefined
```
const GoogleContainer = (props) => {
  const { state } = useContext(GoogleMapContext);
  const [trigger, setTrigger] = useState(0);
  useEffect(() => {
    console.log(trigger, state);
    if (state && state.map) {
      console.log(state);
    }
  }, [trigger]);
  return (
    <ReactGoogleMap
      zoom={4}
      center={{ lat: 37.09024, lng: -95.712891 }}
      opt={{
        scaleControl: false,
      }}
      onZoomChanged={() => { setTrigger(trigger + 1); }}
    >
......
```

**Comment by lucifer1004** [comment]
The problem here is that your `ReactGoogleMap` is not wrapped by `GoogleMapProvider`.

**Comment by lucifer1004** [comment]
You have placed the `GoogleMapProvider` inside `ReactGoogleMap`, which invalidates the context. That is why the state is always undefined.

**Comment by lucifer1004** [comment]
See this [codesandbox](https://codesandbox.io/s/googlemapcontext-example-xvdxk)

**Comment by lucifer1004** [comment]
You can also read the [useContext](https://reactjs.org/docs/hooks-reference.html#usecontext) API for more details.

**Comment by T-Flynn** [comment]
Sorry, my fault. thanks for your patience
